zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -26,6 +26,7 @@ angular.module('zeppelinWebApp').controller('InterpreterCtrl',
     var getInterpreterSettings = function() {
       $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/setting').success(function(data, status, headers, config) {
         $scope.interpreterSettings = data.body;
+        checkDownloadingDependencies();
```

Comment:
This loop `getInterpreterSettings`<> `checkDownloadingDependencies`(this in turn iterates through setting) will be executed until the dependencies are downloaded.
Should be fine?
Otherwise, lgtm